crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -1444,6 +1444,119 @@ impl Store {
     pub fn secrets_stream(&self) -> impl Stream<Item = GossippedSecret> {
         self.inner.store.secrets_stream()
     }
+
+    pub(crate) async fn import_room_keys(
+        &self,
+        exported_keys: Vec<ExportedRoomKey>,
+        #[cfg(feature = "backups_v1")] from_backup: bool,
+        progress_listener: impl Fn(usize, usize),
+    ) -> Result<RoomKeyImportResult> {
+        let mut sessions = Vec::new();
+
+        async fn new_session_better(
```

Comment:
preexisting, but since I've asked for other changes in this PR: since it's called only once, would it make sense to inline this, and store the result into a bool called `is_new_session_better`?

Reply:
It's called once per loop iteration, hmm it makes the loop a bit bigger. Do you still think this is wise? I don't care that much but am not sure if you were aware that it was called once per iteration.